app/src/main/java/cloudcity/LoggingServiceExtensions.java

```diff
@@ -142,20 +142,11 @@ public static void stopCloudCity() {
 
     private static NetworkDataModel getCloudCityData() {
         List<CellInformation> cellsInfo = dp.getCellInformation();
+        List<CellInformation> signalInfo = dp.getSignalStrengthInformation();
         LocationInformation location = dp.getLocation();
-        CellInformation currentCell = null;
 
-        for (CellInformation ci: cellsInfo) {
-            if (!ci.isRegistered()) {
-                continue;
-            }
-
-            currentCell = ci;
-        }
-
-        if (currentCell == null) {
-            return null;
-        }
+        CellInformation currentCell = findRegisteredCell(cellsInfo);
+        CellInformation currentSignal = findRegisteredCell(signalInfo);
```

Comment:
since signalInfo list of CellInformations has only one or zero members, we'll only take the first one if it's non-empty, because it also **isn't** registered...